src/managers/resources/adc.rs

```diff
@@ -0,0 +1,114 @@
+extern crate rppal;
+
+use crate::managers::Manager;
+use crate::managers::ManagerChannelData;
+use crate::managers::Resource;
+use crate::managers::ResourceManager;
+use crate::managers::TASK_SUCCESS;
+use crate::managers::macros::parse_channel_data;
+use crate::request::TaskData::EmgData;
+use crate::sgcp;
+use crate::sgcp::emg::*;
+use crate::todo;
+use anyhow::Error;
+use anyhow::Result;
+use anyhow::anyhow;
+use log::*;
+use rppal::gpio::{Gpio, OutputPin};
+use rppal::spi::{Bus, Mode, SlaveSelect, Spi};
+use std::process::Output;
+use std::{error::Error as StdError, io, io::Write, thread, time::Duration};
+
+
+use tokio::sync::mpsc::channel;
+use tokio::sync::mpsc::Receiver;
+use tokio::sync::mpsc::Sender;
+
+const DEFAULT_BUFFER_SIZE: usize = 100;
+const SPI_DEVICE_PATH: &str = "/dev/spidev0.0";
+const PAUSE_DURATION_MS: u64 = 500; // Pause duration in milliseconds
+
+pub fn process_data(values: Vec<u16>, inner_threshold: u16, outer_threshold: u16) -> Result<i32, Box<dyn StdError>> {
+    if values.len() != 2 {
+        return Err("Expected 2 EMG values".into());
+    }
+
+    if values[0] >= inner_threshold && values[1] <= outer_threshold{
+        Ok(1) // Open
+    } else if values[0] <= inner_threshold && values[1] >= outer_threshold {
+        Ok(0) // Close
+    }
+    else {
+        Ok(-1) // No action
+    }
+}
+
+fn average(list: Vec<u16>) -> Result<u16, &'static str> {
```

Comment:
Same comment here about using `anyhow` errors -- use `Err(anyhow!(<msg>)`, `Err(Error::msg(<msg>))` or `bail!(<msg>)`/`ensure!(<msg>)` when returning errors rather than using `'static str` as the error type.